crates/swc_ecma_transforms_react/src/refresh/mod.rs

```diff
@@ -497,24 +174,24 @@ impl<C: Comments> Refresh<C> {
                 let reg_ident = private_ident!("_c");
                 reg.push((reg_ident.clone(), reg_str));
                 if let Persist::Hoc(hoc) =
-                    self.get_persistent_id_from_possible_hoc(expr, reg, ignore)
+                    self.get_persistent_id_from_possible_hoc(expr, reg, hook_reg)
                 {
-                    *first_arg = Box::new(make_assign_stmt(reg_ident.clone(), first_arg.take()));
-                    if let Some(hook) = &hoc.hook {
-                        let span = call_expr.span;
-                        let first = ExprOrSpread {
+                    let mut first = first_arg.take();
```

Comment:
This will make the output invalid if `hoc.hook` is `None`.

I'm not sure if this is possible, though.

Reply:
It won't, see L194